test/controllers/users_controller_test.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class UsersControllerTest < ActionDispatch::IntegrationTest
+  setup do
+    sign_in users(:serhat)
+    @user = users(:john)
+  end
+
+  test 'should get index' do
+    get users_path
+    assert_response :success
+    assert_select '#add-button', t('users.index.new_user_link')
+  end
+
+  test 'should get show' do
+    get user_path(@user)
+    assert_response :success
+  end
+
+  test 'should get new' do
+    get new_user_path
+    assert_response :success
+  end
+
+  test 'should create user' do
+    assert_difference('User.count') do
+      post users_path params: {
+        user: {
+          id_number: '33998445768',
```

Comment:
Testlere bu şekilde data girmek yerine fixture'ları kullanalım. users fixture'ını zenginleştirebilirsin eğer öyle bir ihtiyaç varsa.

Reply:
Burada user oluşturma işlevi test edildiği için id_number MERNIS test TC'leri arasından olmak şartıyla sahte veri girilebilir.  Bu  sahte verileri [faker](https://github.com/stympy/faker) gem'ini kullanarak oluşturabiliriz.

Eğer fixture kullanırsak zaten sistemde var olan bir kullanıcıyı sisteme tekrardan kayıt etmeye çalışıyormuşuz gibi bir durum oluşacak, bu tür durumları zaten model validasyon testlerinde kontrol ediyoruz fakat önemli controller işlevleri (kullanıcı oluşturma) için ayrıca bu durumlar test etmemizde faydalı olacaktır.

Reply:
`faker` vb. bir başka bağımlılık kurulmasını, daha genel itibariyle her ihtiyaç için GEM kurulmasını doğru bulmuyorum. Şimdilik doğru bir fixture seti ile testlerimizi hızlı çalışır halde tutabiliriz.

> Eğer fixture kullanırsak zaten sistemde var olan bir kullanıcıyı sisteme tekrardan kayıt etmeye çalışıyormuşuz gibi bir durum oluşacak.

Aynı fixture'ı birden çok kez kullanabiliriz. En basiti önce destroy edip sonra controller'a POST etmek gibi.